mne/io/fiff/raw.py

```diff
@@ -117,19 +117,24 @@ def __init__(self, fname, allow_maxshield=False, preload=False,
                 n_samples += r.last_samp - r.first_samp + 1
 
         # Add annotations for in-data skips
-        offsets = [0] + self._raw_lengths[:-1]
-        for extra, first_samp, offset in zip(self._raw_extras,
-                                             self._first_samps, offsets):
-            for skip in extra:
-                if skip['ent'] is None:  # these are skips
-                    if self.annotations is None:
-                        self.set_annotations(Annotations((), (), ()))
-                    start = skip['first'] - first_samp + offset
-                    stop = skip['last'] - first_samp + offset
-                    self.annotations.append(
-                        _sync_onset(self, start / self.info['sfreq']),
-                        (stop - start + 1) / self.info['sfreq'],
-                        'BAD_ACQ_SKIP')
+        for extra in self._raw_extras:
+            start = np.array([_['first'] for _ in extra if _['ent'] is None])
+            stop = np.array([_['last'] for _ in extra if _['ent'] is None])
+            duration = (stop - start + 1) / self.info['sfreq']
```

Comment:
`duration = (stop - start + 1.) / self.info['sfreq']`

I have the habit to force float numerics. Worked too much with python 2 in the past :)